src/arp.c

```diff
@@ -70,6 +71,14 @@ int arp_cache_insert(in_addr_t ip_addr,
     if (ip_addr == 0)
         return 0;
 
+    if ((entry = arp_cache_get_entry(ip_addr)) > 0) {
+        if (memcmp(entry->haddr, haddr, sizeof(mac_addr_t)) != 0) {
```

Comment:
If the contents pointed by `entry->haddr` and `haddr` are the same, why would you duplicate?

Reply:
I duplicate `haddr` to `entry->haddr` when the return value of memcmp not equal to zero. According to `man memcmp` (if I'm not misunderstood) only when two contents are the same that the return value would be zero.

Reply:
Sorry for my last comment (I finds out it's a bit offensive).
Could I have more hints about where should I modify the code ? Thanks.

Reply:
You can consolidate the comments and commit messages for the proposed changes.